libbeat/publisher/queue/diskqueue/queue.go

```diff
@@ -86,6 +91,18 @@ type diskQueue struct {
 	done chan struct{}
 }
 
+// channel request for metrics from an external client
+type metricsRequest struct {
+	response chan metricsRequestResponse
+}
+
+// metrics response from the disk queue
+type metricsRequestResponse struct {
```

Comment:
I think we could add number of items in the queue.  On startup the segments are read, and they contain a count field.  We could update that total when we add & ack events in the queue.

I'm not sure oldestEntryID is very helpful, the ID isn't a time, so if I told you the oldest ID was 13075 what would you do with that information?  I think I'd rather see a guage that shows me the rate things are being added to the queue and being vacated from the queue as well as total number in the queue.

For `OccupiedRead`, is this number of events that we have sent but haven't received an ACK for?

Reply:
> I'm not sure oldestEntryID is very helpful, the ID isn't a time, so if I told you the oldest ID was 13075 what would you do with that information? I think I'd rather see a guage that shows me the rate things are being added to the queue and being vacated from the queue as well as total number in the queue.

I was thinking the same. The only value of oldestEntryID is that we can look to see if it is changing to know if the queue is draining. We will probably get very similar information from observing the current queue size, but the oldestEntryID would let us catch the edge case where the queue is almost always full but data is still moving through it.

I am not opposed to removing this metric, but I think it has some small value.

Reply:
> I think we could add number of items in the queue. On startup the segments are read, and they contain a count field. We could update that total when we add & ack events in the queue

Having `EventCount` for the disk queue would be interesting, and would mean we can always rely on the `EventCount` metric being populated between both queue types which I think makes the metrics easier to use.

Reply:
I'm reluctant to use oldestEntryID as an indicator of a draining queue, if for some reason an event can't be delivered but we have retries, than oldestEntryID might be static but the queue could be draining.